seacrowd/sea_datasets/sredfm/sredfm.py

```diff
@@ -0,0 +1,296 @@
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name="sredfm_source",
+            version=SOURCE_VERSION,
+            description="sredfm source schema",
+            schema="source",
+            subset_id="sredfm",
+        ),
+        SEACrowdConfig(
+            name="sredfm_seacrowd_kb",
+            version=SEACROWD_VERSION,
+            description="sredfm SEACrowd schema",
+            schema="seacrowd_kb",
+            subset_id="sredfm",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "sredfm_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "docid": datasets.Value("string"),
+                    "title": datasets.Value("string"),
+                    "uri": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                    "entities": [
+                        {
+                            "uri": datasets.Value(dtype="string"),
+                            "surfaceform": datasets.Value(dtype="string"),
+                            "type": datasets.Value(dtype="string"),
+                            "start": datasets.Value(dtype="int32"),
+                            "end": datasets.Value(dtype="int32"),
+                        }
+                    ],
+                    "relations": [
+                        {
+                            "subject": datasets.Value(dtype="int32"),
+                            "predicate": datasets.Value(dtype="string"),
+                            "object": datasets.Value(dtype="int32"),
+                        }
+                    ],
+                }
+            )
+
+        elif self.config.schema == "seacrowd_kb":
+            features = schemas.kb_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        urls = _URLS[_DATASETNAME]
+        data_dir = dl_manager.download_and_extract(urls)
+
+        relation_names = dict()
+        with open(_URLS["relations_url"], encoding="utf-8") as f:
+            for row in f:
+                rel_code, rel_name, _, _ = row.strip().split("\t")
+                relation_names[rel_code] = rel_name
```

Comment:
This will download directly from GitHub when examples are generated, instead of loading from a previously donwloaded local file. You could do:
```
path = dl_manager.download(_URLS["relations_url"])
with open(path, encoding="utf-8") as f:
...
```